vignettes/advanced.Rmd

```diff
@@ -268,15 +466,161 @@ pool(anl_obj)
+dat_delta[index_delta, ] <- dat_delta %>%
+    filter(is_missing & is_post_ice) %>%
+    select(-delta) %>%
+    left_join(deltas, by = "visit")
+```
+
+Note that the delta is applied only to observations that were missing and that were after the ICE "treatment discontinuation". 
+
+We can now call `analyse()`, that will first add the delta to the corresponding outcomes and then call the analysis model.
+Be aware that the delta is applied to all imputed datasets:
+
+```{r}
+# Repeat analyses with the delta-adjusted values and pool together the results
+anl_delta <- analyse(impute_obj_CIR, delta = dat_delta, vars = vars_an)
+pool(anl_delta)
+```
+
+## Tipping point analysis
+
+The tipping point analysis is a way of doing sensitivity analysis by repeatedly performing a delta-adjustment with worse and worse value of delta until when the result from the primary analysis is overturned.
+
+In the previous section (\@ref(sec:manualDelta)) we have added a delta to any imputed value after treatment discontinuation proportional to the observed mean change from baseline at the corresponding visit.
+We have set the proportion equal to 20%. We can vary the proportion up to when the treatment effect estimated at the last visit is not significant anymore.
+
+Let's start by writing a function that performs a delta-adjustment based on a proportion of the observed mean change from baseline.
+This function needs as arguments the object resulting from the imputation phase (`impute_obj`), a `data.frame` containing the mean change from baseline at each visit (`mean_chg`) and a number indicating the chosen proportion (`prop`):
+
+```{r}
+delta_adj <- function(impute_obj, mean_chg, prop) {
+
+    # Get delta template dataset
+    dat_delta <- delta_template(imputations = impute_obj)
+
+    # Estimate delta values
+    mean_chg$delta <- prop * mean_chg$delta
+
+    # Select observations corresponding to missing values after the ICE
+    index_delta <- dat_delta$is_missing & dat_delta$is_post_ice
+
+    # Include the value of delta from "mean_chg" to "dat_delta" by joining the two data.frames
+    dat_delta[index_delta, ] <- dat_delta %>%
+        filter(is_missing & is_post_ice) %>%
+        select(-delta) %>%
+        left_join(mean_chg, by = "visit")
+
+    # Repeat analyses with the delta-adjusted values and pool together the results
+    anl_delta <- analyse(impute_obj, delta = dat_delta, vars = vars_an)
+    pool_obj <- pool(anl_delta)
+
+    # Return the results together with the proportion used and the delta values
+    pool_obj$prop <- prop
+    pool_obj$delta <- mean_chg
+    return(pool_obj)
+}
+
+```
+
+To perform the tipping point analysis, we loop over increasing values of the proportion and we stop when the p-value is above the desired confidence level (we assume that it is equal to 0.05).
+
+```{r}
+prop <- seq(from = 0.1, to = 1, by = 0.1)
+
+iter <- 1
+p_val <- 0
+while (p_val <= 0.05 & iter <= length(prop)) {
```

Comment:
I would avoid a while loop, I would simply just run all of the range and then call out the one that tripped the threshold. Only for the sole reason that while loops can be hard to get your head round at times for a reader.

Reply:
Marcel wants to change this as well. I will not resolve this conversation and I will have a look again after his changes.